src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -39,61 +46,75 @@ internal class CallHierarchyCommandHandler : ICommandHandler<ViewCallHierarchyCo
+            IAsynchronousOperationListenerProvider asynchronousOperationListenerProvider,
             [ImportMany] IEnumerable<ICallHierarchyPresenter> presenters,
             CallHierarchyProvider provider)
         {
             _threadingContext = threadingContext;
+            _threadOperationExecutor = threadOperationExecutor;
+            _listener = asynchronousOperationListenerProvider.GetListener(FeatureAttribute.CallHierarchy);
             _presenter = presenters.FirstOrDefault();
             _provider = provider;
         }
 
         public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionContext context)
         {
-            using (var waitScope = context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Computing_Call_Hierarchy_Information))
+            var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
+                context.OperationContext, _threadingContext);
+            if (document == null)
             {
-                var cancellationToken = context.OperationContext.UserCancellationToken;
-                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChanges(
-                    context.OperationContext, _threadingContext);
-                if (document == null)
-                {
-                    return true;
-                }
+                return true;
+            }
+
+            var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
+
+            // We're showing our own UI, ensure the editor doesn't show anything itself.
+            context.OperationContext.TakeOwnership();
+            var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
+            ExecuteCommandAsync(document, caretPosition)
+                .ReportNonFatalErrorAsync()
+                .CompletesAsyncOperation(token);
+
+            return true;
+        }
 
-                var semanticModel = document.GetSemanticModelAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+        private async Task ExecuteCommandAsync(Document document, int caretPosition)
+        {
+            using (var context = _threadOperationExecutor.BeginExecute(
+                ServicesVSResources.Call_Hierarchy, ServicesVSResources.Navigating, allowCancellation: true, showProgress: false))
+            {
+                var cancellationToken = context.UserCancellationToken;
 
-                var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
-                var symbolUnderCaret = SymbolFinder.FindSymbolAtPositionAsync(
-                    semanticModel, caretPosition, document.Project.Solution.Services, cancellationToken)
-                    .WaitAndGetResult(cancellationToken);
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
💡 Since this operation is blocking the UI at a higher level, it would be more efficient (and better practice) to use `ConfigureAwait(true)` throughout this method

Reply:
this op should not be blockign the UI.  Teh caller fires-and-forgets it.  So it would be good to not have any interaction with the UI thread until needed.  For example, jumping back to the UI from this just to go the BG in FindSymbolAtPositionAsync, MapSymbolAsync and the rest isn't necessary :)  

We do need to come back to the UI after all the computation work to actually update hte UI (or display a failure message).  But the coming back should be deferred till then.